docs/DeveloperGuide.md

```diff
@@ -2,12 +2,153 @@
+To load data from previous session:
+Within Wiagi constructor, Storage class is constructed, which will load and initialise incomes, spendings and
+password by de-serialising the text at their distinct file paths. Wiagi will then initialise it incomes and spendings
+based on the member in the Storage class.
+![storageLoad.png](./Diagrams/storageLoad.png)
+
+To save data for current session:
+After the command bye is sent by the user, incomes and spendings will be serialised and overwrite texts in
+their distinct file paths.
+![storageSave.png](./Diagrams/storageSave.png)
+
+### adding of new entry
+![addCommandSequence.jpg](./Diagrams/addCommandSequence.jpg)
+To add new entries, user will have to input the related commands.
+Wiagi will then parse the command to the AddCommand class.
+The AddCommand class will then validate the user's input and add the input to IncomeList or SpendingList
+
+### deleting of entry, editing of entry
+The commands are similar where there would be a parsing of command to each of its individual classes.
+A similar validation process takes place and actions would be made on IncomeList or SpendingList accordingly
+(deleting entry from list for delete and editing of entry from list for edit)
+
+### Recurrence Component
+
+#### Motivation behind the component:</br>
++ Allows the user to set specific expenditure and incomes as recurring events to increase efficiency when using the
+  application
++ Users may have differing frequencies for recurring events thus application gives them a few common options
+
+Illustrated below is the class diagram for the Recurrence Component:</br>
+</br>
+<img src="./Diagrams/recurrenceClassDiagram.png" alt="recurrenceClassDiagram" width="800"/>
+</br>
+</br>
+Illustrated below is the sequence diagram of the Recurrence Component: </br>
+</br>
+<img src="./Diagrams/recurrenceSequenceDiagram.png" alt="recurrenceSequenceDiagram" width="700"/>
+</br>
+For the reference fragment of 'load from storage', refer to [Storage component](#storage). </br>
+For the reference fragment of 'add recurring entry', refer to 
+[checkIncomeRecurrence / checkSpendingRecurrence](#checkincomerecurrence--checkspendingrecurrence-method) method. </br>
+
+
+#### How the Recurrence Component works:</br>
++ Upon running the application by the user, `Storage` component will load the `IncomeList` and `SpendingList` members of
+`Wiagi` to retrieve past data.
++ Both list are then iterated through. Each member of the list is parsed through `Parser` which returns the type of 
+recurrence it is (eg. `DailyRecurrence`, `null`) which is encapsulated as a `Recurrence` object.
++ If `Recurrence` is not `null` (ie. a recurring entry), it checks the entry and adds to the `SpendingList` and 
+`IncomeList` if needed. </br>
+
+#### Implementation:
+#### Recurrence class
+The `Recurrence` class is a abstract class that provides the interface for checking `Income` and `Spending` and adding 
+recurring entries into the list. </br>
```

Comment:
</br> printed in DG webversion
<img width="1315" alt="image" src="https://github.com/user-attachments/assets/6b6eb49e-7637-44d6-b167-ded417e16a6a">